basil/HL/sensirion_sht85.py

```diff
@@ -0,0 +1,179 @@
+        T, RH = self.get_temperature_and_humidity(repeatability)
+        return self.to_dew_point(T, RH)
+
+    def start_asynchronous_read(self, measurments_per_second=1, repeatability=None, ART=False):
+        if ART:
+            cmd = [0x2B, 0x32]
+        else:
+            cmd = {
+                "low"   : {
+                    0.5 : [0x20, 0x2F],
+                    1   : [0x21, 0x2D],
+                    2   : [0x22, 0x2B],
+                    4   : [0x23, 0x29],
+                    10  : [0x27, 0x2A],
+                }[measurments_per_second],
+                "medium": {
+                    0.5 : [0x20, 0x24],
+                    1   : [0x21, 0x26],
+                    2   : [0x22, 0x20],
+                    4   : [0x23, 0x22],
+                    10  : [0x27, 0x21],
+                }[measurments_per_second],
+                "high"  : {
+                    0.5 : [0x20, 0x32],
+                    1   : [0x21, 0x30],
+                    2   : [0x22, 0x36],
+                    4   : [0x23, 0x34],
+                    10  : [0x27, 0x37],
+                }[measurments_per_second]
+            }[self.get_repeatabilaty(repeatability)]
+        self._write(cmd)
+
+    def read_asynchronous(self, timeout_us=0):
+        try:
+            data = self._read([0xE0, 0x00], read_n_words=2, timeout_us=timeout_us)
+            return self.to_temperature(data), self.to_humidity(data)
+        except self.TimeoutError:
+            return None, None
+
+    def stop_asynchronous_read(self):
+        self._write([0x30, 0x93])
+
+    def asynchronous(self, measurments_per_second=1, repeatability=None, ART=False):
+        class Asynchronous:
+            def __enter__(self_a):
+                self.start_asynchronous_read(measurments_per_second, repeatability, ART)
+                return self_a
+            def __exit__(self_a, exc_type, exc_val, exc_tb ):
+                self.stop_asynchronous_read()
+            def read(self_a):
+                return self.read_asynchronous()
+            def read_synchronous(self_a, timeout_us=100e3):
+                return self.read_asynchronous(timeout_us)
+        return Asynchronous()
+
+    def _get_status(self):
+        return self._read([0xF3, 0x2D], read_n_words=1, timeout_us=0)
+    
+    def enable_heater(self):
+        self._write([0x30, 0x6D])
+        
+    def disable_heater(self):
+        self._write([0x30, 0x66])
+
+    def is_heater_on(self):
+        return self._get_status()[0]&(1<<13) > 0
+        
+    def reset_sensor(self):
+        self._write([0x30, 0xA2])
+
+    def get_repeatabilaty(self, repeatability=None):
+        if repeatability is None:
+            return self.repeatability
+        else:
+            return repeatability
+
+    def to_temperature(self, data):
+        return -45+175*(data[0]/(2**16-1))
+
+    def to_humidity(self, data):
+        return 100*(data[1]/(2**16-1))
+
+    def to_dew_point(self, T, RH):
```

Comment:
the sensor doesn't support dew point measurement? I'm not sure if this kind of calculation should be handled by basil.. on the other hand it would be nice to support the same function as other temperature sensors, so they can be interchanged...

Reply:
Unfortunately not. I added it here to suport at least some dew point measurement, since the bdaq53 periphery class defined a getter for the dew point, so it is at least kompatible.